packages/connectivity/src/scp-cf/destination/destination-accessor-jwks.spec.ts

```diff
@@ -0,0 +1,198 @@
+import nock from 'nock';
+import {
+  destinationName,
+  destinationSingleResponse,
+  mockInstanceDestinationsCall,
+  mockServiceBindings,
+  mockServiceToken,
+  mockSingleDestinationCall,
+  mockSubaccountDestinationsCall,
+  mockVerifyJwt,
+  oauthMultipleResponse,
+  providerServiceTokenPayload,
+  signedJwtForVerification,
+  subscriberServiceTokenPayload,
+  subscriberUserPayload
+} from '../../../../../test-resources/test/test-util';
+import * as jwt from '../jwt';
+import { responseWithPublicKey } from '../jwt.spec';
+import { DestinationFetchOptions } from './destination-accessor-types';
+import {
+  alwaysProvider,
+  alwaysSubscriber
+} from './destination-selection-strategies';
+import { getDestination } from './destination-accessor';
+import { DestinationConfiguration } from './destination';
+
+const jku = 'https://my-jku-url.authentication.sap.hana.ondemand.com';
+
+describe('custom jwt via jwks property on destination', () => {
+  beforeEach(() => {
+    mockServiceBindings();
+    mockVerifyJwt();
+    mockServiceToken();
+  });
+
+  afterEach(() => {
+    nock.cleanAll();
+    jest.clearAllMocks();
+  });
+
+  const destFetchOption: DestinationFetchOptions = {
+    destinationName: 'FINAL-DESTINATION',
+    iasToXsuaaTokenExchange: false
+  };
+
+  function mockOneDestination(
+    destination: DestinationConfiguration,
+    serviceToken: string,
+    userJwt: string
+  ) {
+    mockInstanceDestinationsCall(nock, [], 200, serviceToken);
+
+    mockSubaccountDestinationsCall(nock, [destination], 200, serviceToken);
+
+    mockSingleDestinationCall(
+      nock,
+      destinationSingleResponse([destination]),
+      200,
+      destinationName,
+      { authorization: `Bearer ${serviceToken}`, 'x-user-token': userJwt },
+      { badheaders: [] }
+    );
+  }
+
+  it('verifies JWT with JKU property', async () => {
+    nock(jku).get('/').reply(200, responseWithPublicKey());
+    const userJwt = signedJwtForVerification(subscriberUserPayload, jku);
+    const serviceJwt = signedJwtForVerification(
+      subscriberServiceTokenPayload,
+      jku
+    );
+
+    mockOneDestination({ ...oauthMultipleResponse[0] }, serviceJwt, userJwt);
+
+    const spy = jest.spyOn(jwt, 'verifyJwt');
+    const actual = await getDestination({
+      ...destFetchOption,
+      jwt: userJwt,
+      selectionStrategy: alwaysSubscriber
+    });
+    expect(spy).toHaveBeenCalledTimes(1);
+    expect(actual).toBeDefined();
```

Comment:
The type of `actual` is `Destination | null`. Therefore, `.not.toBeNull()` should be used instead of `.toBeDefined()` because `expect(null).toBeDefined()` is also valid. I will fix this and check the other occurrences of `.toBeDefined()`